overlord/snapshotstate/snapshotmgr.go

```diff
@@ -246,7 +246,11 @@ func prepareRestore(task *state.Task) (snapshot *snapshotSetup, oldCfg map[strin
 		}
 	}
 
-	reader, err = backendOpen(snapshot.Filename)
+	ok, setID := backend.IsSnapshotFilename(snapshot.Filename)
```

Comment:
we could have left this behavior in Open itelf by defining and passing backed.ExtractFnameSetID = 0 as SetID to mark for it